File: deendoon/app/Models/ProfessionalCollectionTimelineEvent.php
<?php

namespace App\Models;

use App\Enums\ProfessionalCollectionTimelineEventType;
use Database\Factories\ProfessionalCollectionTimelineEventFactory;
use Illuminate\Database\Eloquent\Attributes\Fillable;
use Illuminate\Database\Eloquent\Concerns\HasUlids;
use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;
use Illuminate\Database\Eloquent\Relations\HasMany;

/**
 * Transfer Case to Deendoon Recovery Team (Product Owner-approved
 * decision): the Deendoon Recovery Team's own operational recovery
 * activity log for a Professional Collection Request — independent of
 * the Audit Log (system/data changes) and the existing, unchanged
 * Customer Collection Timeline. No tenant_id of its own — same pattern
 * as RequestMessage; visibility is derived through the parent Request.
 */
#[Fillable(['professional_collection_request_id', 'event_type', 'officer_user_id', 'occurred_at', 'notes', 'outcome'])]
class ProfessionalCollectionTimelineEvent extends Model
{
    /** @use HasFactory<ProfessionalCollectionTimelineEventFactory> */
    use HasFactory, HasUlids;

    protected function casts(): array
    {
        return [
            'event_type' => ProfessionalCollectionTimelineEventType::class,
            'occurred_at' => 'datetime',
            'created_at' => 'datetime',
            'updated_at' => 'datetime',
        ];
    }

    public function request(): BelongsTo
    {
        return $this->belongsTo(ProfessionalCollectionRequest::class, 'professional_collection_request_id');
    }

    public function officer(): BelongsTo
    {
        return $this->belongsTo(User::class, 'officer_user_id');
    }

    public function attachments(): HasMany
    {
        return $this->hasMany(ProfessionalCollectionRequestAttachment::class, 'timeline_event_id');
    }
}

File: mobile/lib/features/professional_collection/domain/professional_collection_timeline_event.dart
import 'professional_collection_attachment.dart';

/// Mirrors `App\Http\Resources\ProfessionalCollectionTimelineEventResource`
/// exactly — the Deendoon Recovery Team's own operational activity log for
/// a Request, independent of the Audit Log and the Customer Collection
/// Timeline. Business Owner (this app) may only view it —
/// `createTimelineEvent`/`updateTimelineEvent` are Deendoon Platform
/// Administrator-only, so no write path is wrapped anywhere in this
/// feature for this resource.
class ProfessionalCollectionTimelineEvent {
  final String id;
  final String professionalCollectionRequestId;
  final String eventType;
  final String? officerUserId;
  final String occurredAt;
  final String? notes;
  final String? outcome;
  final List<ProfessionalCollectionAttachment> attachments;
  final String createdAt;
  final String updatedAt;

  const ProfessionalCollectionTimelineEvent({
    required this.id,
    required this.professionalCollectionRequestId,
    required this.eventType,
    required this.officerUserId,
    required this.occurredAt,
    required this.notes,
    required this.outcome,
    required this.attachments,
    required this.createdAt,
    required this.updatedAt,
  });

  factory ProfessionalCollectionTimelineEvent.fromJson(Map<String, dynamic> json) =>
      ProfessionalCollectionTimelineEvent(
        id: json['id'].toString(),
        professionalCollectionRequestId: json['professional_collection_request_id'].toString(),
        eventType: json['event_type'] as String,
        officerUserId: json['officer_user_id']?.toString(),
        occurredAt: json['occurred_at'] as String,
        notes: json['notes'] as String?,
        outcome: json['outcome'] as String?,
        attachments: (json['attachments'] as List<dynamic>? ?? [])
            .map((e) => ProfessionalCollectionAttachment.fromJson(e as Map<String, dynamic>))
            .toList(),
        createdAt: json['created_at'] as String,
        updatedAt: json['updated_at'] as String,
      );
}
